lib/search.js

```diff
@@ -0,0 +1,56 @@
+import { trim, curry, cloneDeep, isEmpty } from 'lodash'
+
+export const filter = curry((originalData, setData, searchString) => {
+  searchString = trim(searchString)
+  const includes = (base, target = searchString) => base.includes(target)
+  let data = cloneDeep(originalData)
+
+  if (searchString) {
+    data = data.filter((baseEle, idx) => {
+      let bool = false,
+        temp = false
+
+      if (includes(baseEle.title) || includes(baseEle.description)) return true
+
+      data[idx].workshops = baseEle.workshops.filter(workshop => {
+        temp = false
+
+        if (includes(workshop.name) || includes(workshop.description))
+          (bool = true), (temp = true)
+
+        // name and description don't have a match for target
+        if (!temp) {
+          const filtered = workshop.tags.filter(tag => includes(tag))
+
+          if (filtered.length) (bool = true), (temp = true)
+        }
+        return temp
+      })
+
+      // only required if bool is true
+      if (bool) {
+        data[idx].slugs = baseEle.slugs.filter(slug => {
+          temp = false
+          data[idx].workshops.forEach(workshop => {
+            if (slug == workshop.slug) temp = true
```

Comment:
I'll defer to @MatthewStanciu on code formatting, but want to ping him here b/c this follows different formatting then the rest of the codebase.

Reply:
@bajpai244 feel free to resolve this if he doesn't comment here.

Reply:
I see, lemme run the prettier.config file through my whole codebase to resolve this